File: golang/lc151.go
package main

import "strings"

func reverseWords(s string) string {
	words := strings.Split(strings.Trim(s, " "), " ")

	reversed := make([]string, 0, len(words))
	for i := len(words) - 1; i >= 0; i-- {
		if len(words[i]) == 0 {
			continue
		}
		reversed = append(reversed, words[i])
	}
	return strings.Join(reversed, " ")
}

File: js&ts/LeetCode75/lc-151.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
  let words = s.trim().split(" ");
  let reversed = [];
  for (let i = words.length - 1; i >= 0; i--) {
    if (words[i] === "") {
      continue;
    }
    reversed.push(words[i]);
  }
  return reversed.join(" ");
};
